packages/scripts/src/remove-training-cases.js

```diff
@@ -0,0 +1,354 @@
+	const projectUpdates = await tx.projectUpdate.findMany({
+		where: { caseId }
+	});
+	await Promise.all(
+		projectUpdates.map(async (projectUpdate) => {
+			const { id } = projectUpdate;
+			console.log(`Removing project updates: ${reference}-${id}`);
+			await tx.projectUpdateNotificationLog.deleteMany({ where: { projectUpdateId: id } });
+			await tx.projectUpdate.delete({ where: { id } });
+		})
+	);
+};
+
+/**
+ * @param {import('@prisma/client')} tx
+ * @param caseDetails
+ * @returns {Promise<void>}
+ */
+const removeRepresentations = async (tx, caseDetails) => {
+	const { id: caseId } = caseDetails || {};
+	const representations = await tx.representation.findMany({
+		where: { caseId }
+	});
+	await Promise.all(
+		representations.map(async (representation) => {
+			const { id, reference } = representation;
+			console.log(`Removing representation: ${reference}`);
+			await tx.representationAction.deleteMany({ where: { representationId: id } });
+			await tx.representation.delete({ where: { id } });
+		})
+	);
+};
+
+/**
+ * @param {import('@prisma/client')} tx
+ * @param caseDetails
+ * @returns {Promise<void>}
+ */
+const removeExaminationTimetables = async (tx, caseDetails) => {
+	const { id: caseId, reference } = caseDetails || {};
+	const examinationTimetables = await tx.examinationTimetable.findMany({
+		where: { caseId }
+	});
+	await Promise.all(
+		examinationTimetables.map(async (examinationTimetable) => {
+			const { id } = examinationTimetable;
+			console.log(`Removing examination timetable: ${reference}`);
+			await tx.examinationTimetableItem.deleteMany({ where: { examinationTimetableId: id } });
+			await tx.examinationTimetable.delete({ where: { id } });
+		})
+	);
+};
+
+/**
+ * @param {PrismaClient} tx
+ * @param {any} caseDetails
+ * @returns {Promise<void>}
+ */
+const removeApplicant = async (tx, caseDetails) => {
+	const { applicant, reference } = caseDetails;
+	if (applicant?.id) {
+		console.log(`Removing examination timetable: ${reference}`);
+		const { address } = applicant;
+		if (address?.id) {
+			await tx.address.delete({ where: { id: address.id } });
+		}
+		await tx.serviceUser.delete({ where: { id: applicant.id } });
+	}
+};
+
+/**
+ *
+ * @param {PrismaClient} tx
+ * @param document
+ * @param folderPath
+ * @returns {Promise<void>}
+ */
+const removeDocument = async (tx, document, folderPath) => {
+	const { guid, documentReference } = document;
+
+	console.log(`Removing document: ${folderPath}/${documentReference}`);
+
+	await tx.document.updateMany({
+		where: { guid },
+		data: {
+			latestVersionId: null
+		}
+	});
+
+	await tx.documentVersion.updateMany({
+		where: { documentGuid: guid },
+		data: {
+			transcriptGuid: null
+		}
+	});
+
+	await tx.documentActivityLog.deleteMany({
+		where: {
+			documentGuid: guid
+		}
+	});
+
+	await tx.documentVersion.deleteMany({
+		where: {
+			documentGuid: guid
+		}
+	});
+
+	await tx.document.delete({
+		where: { guid },
+		hardDelete: true
+	});
```

Comment:
We don't actually delete the document in blob. I think we either need to add that, or output a list of blobs that can be safely deleted.

Reply:
I've added the list of blobs to the script where the document version is deleted